metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/hydrator/ChartHydrator.java

```diff
@@ -1,45 +1,35 @@
 package com.linkedin.metadata.kafka.hydrator;
 
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.linkedin.chart.ChartInfo;
-import com.linkedin.common.urn.ChartUrn;
 import com.linkedin.metadata.aspect.ChartAspect;
-import com.linkedin.metadata.dao.RestliRemoteDAO;
-import com.linkedin.metadata.snapshot.ChartSnapshot;
-import com.linkedin.restli.client.Client;
-import java.net.URISyntaxException;
-import java.util.Optional;
+import com.linkedin.metadata.key.ChartKey;
+import com.linkedin.metadata.snapshot.Snapshot;
+import lombok.experimental.SuperBuilder;
 import lombok.extern.slf4j.Slf4j;
 
 
 @Slf4j
-public class ChartHydrator implements Hydrator {
-  private final Client _restliClient;
-  private final RestliRemoteDAO<ChartSnapshot, ChartAspect, ChartUrn> _remoteDAO;
+@SuperBuilder
+public class ChartHydrator extends Hydrator<ChartKey> {
 
   private static final String DASHBOARD_TOOL = "dashboardTool";
   private static final String TITLE = "title";
 
-  public ChartHydrator(Client restliClient) {
-    _restliClient = restliClient;
-    _remoteDAO = new RestliRemoteDAO<>(ChartSnapshot.class, ChartAspect.class, _restliClient);
+  @Override
+  protected void hydrateFromKey(ObjectNode document, ChartKey key) {
+    document.put(DASHBOARD_TOOL, key.getDashboardTool());
   }
 
   @Override
-  public Optional<ObjectNode> getHydratedEntity(String urn) {
-    ChartUrn chartUrn;
-    try {
-      chartUrn = ChartUrn.createFromString(urn);
-    } catch (URISyntaxException e) {
-      log.info("Invalid Chart URN: {}", urn);
-      return Optional.empty();
+  protected void hydrateFromSnapshot(ObjectNode document, Snapshot snapshot) {
```

Comment:
What if we typed this class on "ChartSnapshot" and actually passed in a ChartSnapshot here?

Reply:
This is the issue I talked about. If there's a way to generically cast the snapshot into chartsnapshot, yes. Otherwise, I will need another abstract function that converts snapshot into ChartSnapshot